src/EditorFeatures/CSharp/SplitStringLiteral/SplitStringLiteralCommandHandler.cs

```diff
@@ -110,27 +112,28 @@ public bool ExecuteCommandWorker(ReturnKeyCommandArgs args)
             return true;
         }
 
-        private bool SplitString(ITextView textView, ITextBuffer subjectBuffer, int position, bool useTabs, int tabSize, CancellationToken cancellationToken)
+        private bool SplitString(ITextView textView, ITextBuffer subjectBuffer, int position, ref IndentationOptions? lazyOptions, CancellationToken cancellationToken)
         {
             var document = subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
             if (document == null)
             {
                 return false;
             }
 
-            // TODO: read option from textView.Options (https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1412138)
-            var options = document.GetIndentationOptionsAsync(_globalOptions, cancellationToken).WaitAndGetResult(cancellationToken);
+            lazyOptions ??= subjectBuffer.GetIndentationOptions(_editorOptionsFatory, _globalOptions, document.Project.LanguageServices);
```

Comment:
is it really important for this to be lazy? (not arguing, just curious).

Reply:
Avoids reading the options for each span. IDK how important scenario is it. I think this method can be generally improved - we apply change for every span, one by one. Instead we could just accumulate all changes and apply them at once. I did not want to broaden the scope of the PR that much.